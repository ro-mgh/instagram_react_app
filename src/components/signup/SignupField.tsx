import React, { FunctionComponent } from "react";
import Signup from "./Signup";
import link1 from "../../pictures/link_signup_appstore.png";
import link2 from "../../pictures/link_signup_googlestore.png";
import { Link } from "react-router-dom";

const SignupField: FunctionComponent = () => {
  return (
    <div className="div-sign-form">
      <div className="sign-mainfield">
        <div className="customlogo-wrapper-sign">
          <h1 className="sign-logo sign-up-logo">Insta</h1>
          <div className="customlogo-div-sign"> | by RM</div>
        </div>
        <h2 className="sign-up-text">
          Sign up to see photos and videos from your friends.
        </h2>
        <button className="sign-up-loginFacebook">
          <span className="sign-up-loginFacebook-logo"></span>
          Log in with Facebook
          {/* <span className="sign-loginFacebook-text">Log in with Facebook</span> */}
        </button>
        <div className="sign-or">
          <div className="sign-or-dash"></div>
          <div className="sign-or-text">or</div>
          <div className="sign-or-dash"></div>
        </div>
        <Signup />
        <p className="sign-up-termsp">
          By signing up, you agree to our
          <a
            className="sign-up-terms"
            target="_blank"
            rel="noreferrer"
            href="https://help.instagram.com/581066165581870"
            tabIndex={0}
          >
            {" "}
            Terms
          </a>
          ,{" "}
          <a
            className="sign-up-terms"
            target="_blank"
            rel="noreferrer"
            href="https://help.instagram.com/519522125107875"
            tabIndex={0}
          >
            {" "}
            Data Policy{" "}
          </a>{" "}
          and{" "}
          <a
            className="sign-up-terms"
            target="_blank"
            rel="noreferrer"
            href="https://help.instagram.com/519522125107875"
            tabIndex={0}
          >
            {" "}
            Cookies Policy
          </a>
          .
        </p>
      </div>
      <div className="sign-redirect">
        <p className="sign-redirect-p">
          Have an account?{" "}
          <Link to="/signin">
            <div className="sign-redirect-a">
              <span className="sign-redirect-span">Log in</span>
            </div>
          </Link>
        </p>
      </div>
      <p className="sign-links-p">Get the app.</p>
      <div className="sign-links-div">
        <img className="sign-links-img" src={link1} alt=""></img>
        <img className="sign-links-img" src={link2} alt=""></img>
      </div>
    </div>
  );
};

export default SignupField;
