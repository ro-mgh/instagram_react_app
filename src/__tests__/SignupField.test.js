import { unmountComponentAtNode } from "react-dom";
import React from "react";
import { render, screen } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

import configureStore from "redux-mock-store";

import { Provider } from "react-redux";

import SignupField from "../components/signup/SignupField";
// import ProtectedSignup from "../views/ProtectedSignup";

const mockStore = configureStore([]);

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("rendering Signup component", () => {
  let store;
  let component;
  const history = createMemoryHistory();

  it("renders Signup component without user", () => {
    store = mockStore({
      firebaseReducer: {
        auth: {
          isLoaded: true,
          isEmpty: true,
        },
      },
      authReducer: {
        authMsgSuccess: "",
        authMsgError: "",
      },
    });

    component = render(
      <Provider store={store}>
        <Router history={history}>
          <SignupField />
        </Router>
      </Provider>
    );
    // screen.debug();
    expect(component.container).toMatchSnapshot();
    expect(screen.getByTestId("signup")).toBeInTheDocument();
  });
});
