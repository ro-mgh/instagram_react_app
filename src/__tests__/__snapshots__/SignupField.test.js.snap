// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering Signup component renders Signup component without user 1`] = `
<div>
  <div
    class="div-sign-form"
  >
    <div
      class="sign-mainfield"
    >
      <div
        class="customlogo-wrapper-sign"
      >
        <h1
          class="sign-logo sign-up-logo"
        >
          Insta
        </h1>
        <div
          class="customlogo-div-sign"
        >
           | by RM
        </div>
      </div>
      <h2
        class="sign-up-text"
      >
        Sign up to see photos and videos from your friends.
      </h2>
      <button
        class="sign-up-loginFacebook"
      >
        <span
          class="sign-up-loginFacebook-logo"
        />
        Log in with Facebook
      </button>
      <div
        class="sign-or"
      >
        <div
          class="sign-or-dash"
        />
        <div
          class="sign-or-text"
        >
          or
        </div>
        <div
          class="sign-or-dash"
        />
      </div>
      <div
        class="sign-form-div"
        data-testid="signup"
      >
        <form
          class="sign-form"
          novalidate=""
        >
          <div
            class="sign-input-field"
          >
            <label
              class="sign-input-label"
            >
              <span
                class="sign-input-span"
              >
                Mobile Number or Email
              </span>
              <input
                aria-label="Mobile Number or Email"
                aria-required="true"
                autocapitalize="off"
                autocorrect="off"
                class="sign-input-form"
                maxlength="75"
                name="email"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="sign-input-field"
          >
            <label
              class="sign-input-label"
            >
              <span
                class="sign-input-span"
              >
                Full Name
              </span>
              <input
                aria-label="Full Name"
                aria-required="true"
                autocapitalize="off"
                autocorrect="off"
                class="sign-input-form"
                maxlength="75"
                name="name"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="sign-input-field"
          >
            <label
              class="sign-input-label"
            >
              <span
                class="sign-input-span"
              >
                Username
              </span>
              <input
                aria-label="Username"
                aria-required="true"
                autocapitalize="off"
                autocorrect="off"
                class="sign-input-form"
                maxlength="75"
                name="username"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="sign-input-field"
          >
            <label
              class="sign-input-label"
            >
              <span
                class="sign-input-span"
              >
                Password
              </span>
              <input
                aria-label="Password"
                aria-required="true"
                autocapitalize="off"
                autocorrect="off"
                class="sign-input-form"
                name="password"
                type="password"
                value=""
              />
            </label>
            <div />
          </div>
          <div
            class="sign-button-div"
          >
            <button
              class="sign-button-submit"
            >
              Log in
            </button>
          </div>
        </form>
      </div>
      <p
        class="sign-up-termsp"
      >
        By signing up, you agree to our
        <a
          class="sign-up-terms"
          href="https://help.instagram.com/581066165581870"
          rel="noreferrer"
          tabindex="0"
          target="_blank"
        >
           
          Terms
        </a>
        ,
         
        <a
          class="sign-up-terms"
          href="https://help.instagram.com/519522125107875"
          rel="noreferrer"
          tabindex="0"
          target="_blank"
        >
           
          Data Policy
           
        </a>
         
        and
         
        <a
          class="sign-up-terms"
          href="https://help.instagram.com/519522125107875"
          rel="noreferrer"
          tabindex="0"
          target="_blank"
        >
           
          Cookies Policy
        </a>
        .
      </p>
    </div>
    <div
      class="sign-redirect"
    >
      <div
        class="sign-redirect-p"
      >
        Have an account?
         
        <a
          href="/signin"
        >
          <div
            class="sign-redirect-a"
          >
            <span
              class="sign-redirect-span"
            >
              Log in
            </span>
          </div>
        </a>
      </div>
    </div>
    <p
      class="sign-links-p"
    >
      Get the app.
    </p>
    <div
      class="sign-links-div"
    >
      <img
        alt=""
        class="sign-links-img"
        src="link_signup_appstore.png"
      />
      <img
        alt=""
        class="sign-links-img"
        src="link_signup_googlestore.png"
      />
    </div>
  </div>
</div>
`;
